### PR TITLE
METRON-1567: Large error message can't be written in Solr

### PR DESCRIPTION
**Pull request by justinleet**
## Contributor Comments
This PR is against the feature branch.

There's a hard limit of ~32kb on string fields.  This migrates the raw_message fields to an unanalyzed TextField type, along with making it a dynamic field to handle the split into raw_messages_<number>.

An integration test is added for this behavior which ensures no error is thrown on huge text, ensures the multiple values work as expected the say way, and that a plain string field does actually cause problems.

## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  

In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? If not one needs to be created at [Metron Jira](https://issues.apache.org/jira/browse/METRON/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel).
- [x] Does your PR title start with METRON-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

### For code changes:
- [x] Have you included steps to reproduce the behavior or problem that is being changed or addressed?
- [x] Have you included steps or a guide to how the change may be verified and tested manually?
- [ ] Have you ensured that the full suite of tests and checks have been executed in the root metron folder via:
  ```
  mvn -q clean integration-test install && dev-utilities/build-utils/verify_licenses.sh 
  ```

- [x] Have you written or updated unit tests and or integration tests to verify your changes?
- [ ] Have you verified the basic functionality of the build by building and running locally with Vagrant full-dev environment or the equivalent?

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.

### CODE REVIEW

**Comment by cestella** [comment]
+1 by inspection

**Comment by justinleet** [comment]
Closing manually because feature branch

File: metron-platform/metron-solr/src/main/config/schema/error/schema.xml [hunk 1]
```diff
@@ -40,6 +39,8 @@
   <field name="guid" type="string" indexed="true" stored="true" required="true" multiValued="false" />
   <uniqueKey>guid</uniqueKey>
 
+  <!-- Raw messages can be split into multiple fields -->
+  <dynamicField name="raw_message*" type="text_raw" indexed="false" stored="true"/>
```
Comment (simonellistonball) [inline metron-platform/metron-solr/src/main/config/schema/error/schema.xml]: Out of curiosity, under which circumstances do we get multiple raw messages?

Comment (justinleet) [reply metron-platform/metron-solr/src/main/config/schema/error/schema.xml]: When the BulkWriterComponent hits an error, there's aggregation of the messages of the Tuples involved.

See:
https://github.com/apache/metron/blob/f072ed231f9acdf0957f0244b6402b2ecd9403a0/metron-platform/metron-writer/src/main/java/org/apache/metron/writer/BulkWriterComponent.java#L117

And:
https://github.com/apache/metron/blob/358e40bd17d5de2a380a91115eaee9ebd3a453f6/metron-platform/metron-common/src/main/java/org/apache/metron/common/error/MetronError.java#L147

Comment (simonellistonball) [reply metron-platform/metron-solr/src/main/config/schema/error/schema.xml]: Makes sense, +1 on this then.

**Pull request closed without merging**